ext/standard/http_fopen_wrapper.c

```diff
@@ -197,9 +198,10 @@ static php_stream *php_stream_url_wrap_http_ex(php_stream_wrapper *wrapper,
 			Z_STRLEN_P(tmpzval) > 0) {
 			use_proxy = 1;
 			transport_len = Z_STRLEN_P(tmpzval);
-			transport_string = estrndup(Z_STRVAL_P(tmpzval), Z_STRLEN_P(tmpzval));
+			transport_string = Z_STRVAL_P(tmpzval);
 		} else {
-			transport_len = spprintf(&transport_string, 0, "%s://%s:%d", use_ssl ? "ssl" : "tcp", ZSTR_VAL(resource->host), resource->port);
+			transport_len = spprintf(&_transport_string, 0, "%s://%s:%d", use_ssl ? "ssl" : "tcp", ZSTR_VAL(resource->host), resource->port);
+			transport_string = _transport_string;
```

Comment:
Maybe it would make more sense to store transport_string as zend_string instead? Then you could use zend_string_copy above and zend_strpprintf here.

Reply:
too young too naive 😵